### ISSUE TITLE
Allow adding custom labels to the "gpu-operator" ServiceMonitor

### ISSUE THREAD

**Issue by peihsuant** [NVIDIA/gpu-operator#710]
### feature description
There is a ServiceMonitor named `gpu-operator` that was automatically created and is owned by the ClusterPolicy. I would like to add custom labels to it so the Prometheus instance with a serviceMonitorSelector can scrape it.

Thanks.

**Comment by cdesiniotis** [comment]
@peihsuant this is already possible today with the `dcgmExporter.serviceMonitor.additionalLabels` field in clusterpolicy.

**Comment by peihsuant** [comment]
Hi @cdesiniotis, what I need to modify is the servicemonitor of `gpu-operator` not `dcgmExporter`, thanks.

**Comment by cdesiniotis** [comment]
@peihsuant apologies, I misread the description. I see we do not have any fields for configuring the `gpu-operator` ServiceMonitor, unlike for `dcgmExporter`. 

If you are interested in working on this, PRs against our gitlab repository are always welcome. https://gitlab.com/nvidia/kubernetes/gpu-operator

**Comment by csauoss** [comment]
@cdesiniotis I would like to give this a try if its still needed.

**Comment by cdesiniotis** [comment]
@csauoss yes this is open. PRs are welcome here https://github.com/NVIDIA/gpu-operator/pulls

**Comment by csauoss** [comment]
@cdesiniotis thank you, but should I create a MR in gitlab(since documentation mentions it) or a PR here is fine as well?

**Comment by csauoss** [comment]
@cdesiniotis I created [MR 1099 in gitlab](https://gitlab.com/nvidia/kubernetes/gpu-operator/-/merge_requests/1099) for this issue for now.  Please review it when you get a chance.

**Comment by cdesiniotis** [comment]
@csauoss we have recently migrated to GitHub and now perform development here. Can you open you PR against https://github.com/NVIDIA/gpu-operator?

> (since documentation mentions it)

Can you point me to where our documentation states this?

**Comment by csauoss** [comment]
Thanks @cdesiniotis! sounds good, will PR here in that case. Here in [CONTRIBUTING.md](https://github.com/NVIDIA/gpu-operator/blob/main/CONTRIBUTING.md#-source-code) file is where I saw references to Gitlab.

**Comment by csauoss** [comment]
> Thanks @cdesiniotis! sounds good, will PR here in that case. Here in [CONTRIBUTING.md](https://github.com/NVIDIA/gpu-operator/blob/main/CONTRIBUTING.md#-source-code) file is where I saw references to Gitlab.

Actually the [pr template in github too](https://github.com/NVIDIA/gpu-operator/blob/main/.github/pull_request_template.md)

**Comment by cdesiniotis** [comment]
Thank you for pointing this out. I have filed https://github.com/NVIDIA/gpu-operator/pull/851 to update our docs.

**Comment by csauoss** [comment]
@cdesiniotis I created pr  #850 in github to address this issue. Please review it when you get a chance. Thanks!

**Comment by csauoss** [comment]
@awrel34 thanks for bumping it. PR #850 is awaiting review.

**Comment by icedream** [comment]
I'm also looking for this feature but noticed the PR #850 was automatically closed.

**Comment by rahulait** [comment]
I've re-opened it as it was autoclosed.

**Comment by rahulait** [comment]
Closing this as this has been fixed/implemented in https://github.com/NVIDIA/gpu-operator/pull/850. This will be available starting v26.7.0 release
